persist/fs/persist_manager.go

```diff
@@ -248,6 +263,26 @@ func (pm *persistManager) Prepare(
 	return prepared, nil
 }
 
+func (pm *persistManager) filesetExistsAt(prepareOpts persist.PrepareOptions) (bool, error) {
+	var (
+		blockStart   = prepareOpts.BlockStart
+		shard        = prepareOpts.Shard
+		snapshotTime = prepareOpts.SnapshotTime
+		nsID         = prepareOpts.NamespaceMetadata.ID()
+	)
+
+	switch prepareOpts.FilesetType {
+	case persist.FilesetSnapshotType:
+		return SnapshotFilesetExistsAt(pm.filePathPrefix, nsID, shard, snapshotTime)
```

Comment:
`SnapshotFilesetExistsAt` assumes the time is the block start, but you're passing it a snapshot time. Is that correct? 

```
func SnapshotFilesetExistsAt(prefix string, namespace ident.ID, shard uint32, blockStart time.Time) (bool, error) ...
```

How do you feel about a type alias for snapshot time - `type SnapshotTime Time.Time` somewhere. It'd catch these kinda issues more easily.

Reply:
Actually now that I'm thinking about this, I think we can probably just delete this code. Snapshots aren't like flushes, we have duplicates by nature. Not a whole lot of point in checking if the files already exists, because the logic for writing out a new fileset (embedded in the writer itself) is already:

1. Check what the latest snapshot file index is
2. Write that index+1

So checking for duplicates is kind of useless.

Gonna delete this check if you're cool with it. Also gonna probably delete the `SnapshotFilesetExistsAt` function because I already have a `LatestForBlock` helper method

Reply:
done